### ISSUE TITLE
Where does the windows application stores the data?

### ISSUE THREAD

**Issue by stevleibelt** [kosmas58/pia-app#98]
Hello,

thank you for your work @kosmas58.

Today a member of our staff came to me and was astonished by the fact that all the created DSFA where gone.
The staff has to use your tool once in a while, meaning a pause of three and more months between two uses is normal.
Right now, I've started the tool on my own and created a dummy DSFA. If I am searching through all files below my windows user path (`c:/user/<username>`), I can not find a file containing my created DSFA.
Is there a place where I can look closer?

Do you have any idea why the described scenario of "just all DSFA-entries are gone" can happen?

Best regards and thanks in advance,
Stev

**Comment by kosmas58** [comment]
`C:\Users\<username>\AppData\Roaming\PIA` is the place where the app stores the data.

Best regards,
Kosmas

**Comment by stevleibelt** [comment]
Hello @kosmas58 and thanks for your quick reply.

I've created an example entry and found this path too, but if I search for a phrase I've used in my example entry, I can not find it.
For searching I am using the commandline tool `grep`.
I found the `databases\Databases.db`. But this sqlite database does not contain any data (Opened it with `sqlite3.exe databases\Databases.db` and ran a `SELECT * FROM databases`).

Hopefully I won't make you mad by (ab)-using this issue ticket raising another question.
The person who installed `pia` on the machine for our staff is not available anymore. Your windows releases are marked as `beta`. Do you plan to release a `stable` version?
Or is it kind of common to just upgrade to the latest beta version? Our current installation is a `1.6.3`.

Best regards and, again, thanks in advance,
Stev

# edit

The only place where I found content is in the path `C:\Users\<user name>\AppData\Roaming\PIA\IndexedDB\file__0.indexeddb.leveldb\000003.log`.
Am I able to rebuild entries from this?

**Comment by kosmas58** [comment]
You'e right. The data are stored in IndexedDB. I can see them in my version of the pia-app through `Toogle Dev Tools`.

The problem of vanishing PIAs occurs when you use different versions of the tool. You have to export the pias as
 *.json files in the old version of the tool and then you have to import them in the new version of the tool.

**Comment by kosmas58** [comment]
My version 2.4 won't become stable, because LINcnil is planning a complete overhaul of the PIA tool (https://github.com/LINCnil/pia/issues/503#issue-720015334).

**Comment by stevleibelt** [comment]
Thank yo @kosmas58, you helped me a lot having a good "kickstart" into this application.

Before closing this issue, I was not able to solve my initial question. The `00003.log` file contains a mix of binary and text data. I was able to fetch the text content to restore, at least, a bit of content.
I tried out some npm based applications or python libraries but nothing was really good to easily open an indexeddb/leveldb database from the filesystem and to export it in a csv or json or whatever.
